packages/fuels-programs/src/contract.rs

```diff
@@ -476,6 +479,29 @@ where
             .map_err(|err| map_revert_error(err, &self.log_decoder))
     }
 
+    pub async fn call_with_retry(
+        mut self,
+        max_attempts: Option<u64>,
+    ) -> Result<FuelCallResponse<D>> {
+        let attempts = max_attempts.unwrap_or(0);
+        let mut delay = Duration::from_secs(1);
+
+        for _ in 1..=attempts {
+            match self.call_or_simulate(false).await {
+                Ok(response) => return Ok(response),
+                Err(ProviderError(description))
+                    if description.starts_with("Client request error") =>
```

Comment:
I think we're better off matching with the `ClientRequestError` variant rather than a string description that can change.

Reply:
Yep. I'd leave a match statement that retries only if the provider error was network related. 

Not sure how fine-tuned you can make that. For example, schema incompatibility between our gql client and the server might also appear as the same type of error (since fuel gql client uses `anyhow`). I suggest digging a bit to see whether you can make any sort of robust claim as to the transitive nature of the error.

You can always send a PR over to the client for more library-friendly errors.

Reply:
You're right, but this is currently the "only" way to do matching. I should talk to the client team.